src/answers.sql

```diff
@@ -1,15 +1,290 @@
+FROM 
+    accounts a
+INNER JOIN 
+    movements m
+ON 
+    a.id IN (m.account_from, m.account_to)
+WHERE 
+    a.id IN ('3b79e403-c788-495a-a8ca-86ad7643afaf', 'fd244313-36e5-4a17-a27c-f8265bc46590')
+GROUP BY 
+    a.id, a.mount;
+
+
+
+
+--(B) Add a new movement with the information: from: 3b79e403-c788-495a-a8ca-86ad7643afaf make a transfer to fd244313-36e5-4a17-a27c-f8265bc46590 mount: 50.75
+DO $$ 
+DECLARE
+    first_account UUID := '3b79e403-c788-495a-a8ca-86ad7643afaf';
+    second_account UUID := 'fd244313-36e5-4a17-a27c-f8265bc46590';
+    transfer_amount NUMERIC := 80;
+BEGIN
+
+    UPDATE accounts
+    SET mount = mount - transfer_amount
+    WHERE id = first_account;
+
+    UPDATE accounts
+    SET mount = mount + transfer_amount
+    WHERE id = second_account;
+
+    INSERT INTO movements (id, type, account_from, account_to, mount, created_at, updated_at)
+    VALUES (
+        gen_random_uuid(),  
+        'TRANSFER',         
+        first_account,      
+        second_account,     
+        transfer_amount,    
+        NOW(),              
+        NOW()               
+    );
+
+    RAISE NOTICE 'Transferencia completada de % a % con un monto de %', 
+                 first_account, second_account, transfer_amount;
+END $$;
+
+
+
+
+-----------------------------------------
+--Shows the result of the tranfer--
+-----------------------------------------
+SELECT 
+    m.type, 
+    m.account_from, 
+    m.account_to, 
+    m.mount AS transferred_amount
+FROM movements m
+WHERE m.account_from = '3b79e403-c788-495a-a8ca-86ad7643afaf'
+   OR m.account_to = '3b79e403-c788-495a-a8ca-86ad7643afaf'
+ORDER BY m.created_at ASC;
+
+
+
+--(c) Add a new movement with the information: from: 3b79e403-c788-495a-a8ca-86ad7643afaf type: OUT mount: 731823.56
+DO $$ 
+DECLARE
+    current_balance NUMERIC;
+	first_account UUID:= '3b79e403-c788-495a-a8ca-86ad7643afaf';
+	out_amount NUMERIC := 731823.56;
+BEGIN
+
+    SELECT a.mount INTO current_balance
+    FROM accounts a
+    WHERE a.id = first_account;
+
+    IF current_balance < out_amount THEN
+        RAISE EXCEPTION 'Saldo insuficiente para realizar el movimiento, transacción rechazada';
+    END IF;
+
+    INSERT INTO movements (id, type, account_from, account_to, mount, created_at, updated_at)
+    VALUES (
+        gen_random_uuid(),
+        'OUT', 
+        first_account,
+		NULL,
+		out_amount,
+        now(),
+        now()
+    );
+	
+EXCEPTION
+WHEN OTHERS THEN
+    RAISE NOTICE '*** Error procesando la transacción con monto %. Saldo disponible: % % ** Haciendo ROLLBACK **', 
+                 out_amount, current_balance, CHR(10);
+    ROLLBACK;
+    RETURN;
+
+END $$;
+
+--(d). Put your answer here if the transaction fails(YES/NO): Yes  the  is failed cuz the mount is higher than the current balance 
+
+--(e and f). If the transaction fails, make the correction on step c to avoid the failure (the way to avoid the failure is tranfer less money than the current balance)
```

Comment:
The solution will change based on my comment to wrap the whole exercise in a transaction.